homeassistant/components/shopping_list/__init__.py

```diff
@@ -53,20 +54,32 @@
 )
 
 
-@asyncio.coroutine
-def async_setup(hass, config):
+async def async_setup(hass, config):
     """Initialize the shopping list."""
 
-    @asyncio.coroutine
-    def add_item_service(call):
+    if DOMAIN not in config:
+        return True
+
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": config_entries.SOURCE_IMPORT}
+        )
+    )
+
+    return True
+
+
+async def async_setup_entry(hass, config_entry=None):
```

Comment:
I don't think we should be able to pass `None` as config_entry.
```suggestion
async def async_setup_entry(hass, config_entry):
```